src/grid/GridLoaderContainer.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { useState, useEffect } from 'react';
+import { RouteComponentProps } from 'react-router-dom';
+import { GridContainerProps } from './GridContainer';
+import firebase from '../firebase/firebase';
+import { GridLoader } from './GridLoader';
+
+export interface NewGridProps {
+  rows: number;
+  cols: number;
+  content: string[];
+}
+
+const GridLoaderContainer: React.FunctionComponent<RouteComponentProps> = (
+  props: RouteComponentProps,
+) => {
+  const [gridProps, setGridProps] = useState<GridContainerProps>();
+  const [loading, setLoading] = useState<boolean>(true);
+
+  const { location } = props;
+  const gridId: string = window.location.pathname.replace('/', '');
+  const newGridProps: NewGridProps = location ? location.state : undefined;
+
+  useEffect(() => {
+    const loadGrid = async () => {
+      const userId = await firebase.login();
+      const grid = await firebase.loadGrid(gridId, userId);
+
+      if (grid) {
+        setLoading(false);
+        setGridProps(grid);
+      } else if (newGridProps) {
```

Comment:
For me, I think this logic would be best placed in `src/create/CreateContainer.tsx`, as its more closely related.

If you were to do this, `CreateContainer` could still provide the new Grid's props to this component via router props

Reply:
Yeah I was a little split on this. Advantage of having it in a Loader component is that all the loading/creating is handled in the same place and it's easy to avoid the glitchy create then load scenario. I'll see if I can tidy that up a different way though

Reply:
Have some ideas, let's catch-up tomorrow!